lib/node-utils.ts

```diff
@@ -211,11 +197,16 @@ export function hasChainedThen(node: TSESTree.Node): boolean {
   return hasThenProperty(parent);
 }
 
+export function isPromiseIdentifier(
```

Comment:
Nice, I like this one 👍

Reply:
I got the inspiration for typing it from ts-eslint itself, it's a really cool way of checking specific node types plus additional checks.